tl/features/context_match.py

```diff
@@ -0,0 +1,332 @@
+import pandas as pd
+import re
+import sys
+import os
+
+class match(object):
+    def __init__(self, input_path, context_path, args):
+        #print(args)
+        self.final_data = pd.read_csv(input_path)
+        self.data = pd.DataFrame()
+        self.result_data = pd.DataFrame()
+        self.context = pd.read_csv(context_path)
+        self.output_column_name = args.get("output_column")
+        self.threshold_1 = args.pop("sim_string")
+        self.threshold_2 = args.pop("sim_quantity")
+        self.debug = args.pop("debug")
+        self.final_data = self.final_data.reindex(columns = self.final_data.columns.tolist() 
+                                  + ['context_properties', 'monge_elkan_sim', self.output_column_name])
+    def preprocess(self, sent):
+        sent = sent.lower()
+        new_sent = self.remove_punctuation(sent)
+        new_sent = self.tokenize(new_sent)
+        return new_sent
+
+    def tokenize(self, sent):
+        sent = sent.split(" ")
+        return sent
+
+    def remove_punctuation(self, sent):

```

Comment:
what punctuations are you removing? Use https://github.com/usc-isi-i2/table-linker/blob/development/tl/preprocess/preprocess.py#L123 instead

Reply:
Hi, depending on the context, I sometimes have to keep the punctuations in place, in order to match. Example - Apple, Samsung|4.88 needs to be kept as it is, so that I can look for both Apple and Samsung Separately and then check if they match the same wikidata properties. And for quantity matching 4.88 is important.